File: frontend/src/types/transaction.ts
export type TransactionType = 'income' | 'expense';

export interface Transaction {
  id?: string;
  type: TransactionType;
  description: string;
  amount: number;
  due_date: string;
  competence_date: string;
  account_id: string;
  category_id: string;
  status: 'Pago' | 'Pendente';
  is_recurring?: boolean;
  observation?: string;
  created_at?: string;
  updated_at?: string;
  is_paid?: boolean;
}

export interface CreateTransactionRequest {
  type: TransactionType;
  description: string;
  amount: number;
  due_date: string;
  competence_date: string;
  account_id: string;
  category_id: string;
  is_recurring?: boolean;
  observation?: string;
  is_paid?: boolean;
} 

File: backend/structs/transaction.go
package structs

import "time"

type Transaction struct {
	ID                  string     `json:"id"`
	UserID              string     `json:"user_id"`
	Description         string     `json:"description"`
	Amount              int        `json:"amount"` // INTEGER no banco
	Type                string     `json:"type"`
	CategoryID          string     `json:"category_id"`
	AccountID           string     `json:"account_id"`
	DueDate             time.Time  `json:"due_date"`
	CompetenceDate      time.Time  `json:"competence_date"`
	IsPaid              bool       `json:"is_paid"`
	Observation         string     `json:"observation"`
	IsRecurring         bool       `json:"is_recurring"`
	RecurringType       *string    `json:"recurring_type"`
	Installments        int        `json:"installments"`
	CurrentInstallment  int        `json:"current_installment"`
	ParentTransactionID *string    `json:"parent_transaction_id"`
	TransferID          *string    `json:"transfer_id"`
	CreatedAt           time.Time  `json:"created_at"`
	UpdatedAt           time.Time  `json:"updated_at"`
	DeletedAt           *time.Time `json:"deleted_at,omitempty"`
}
